File: backend/src/main/java/com/petcare/model/Vaccination.java
package com.petcare.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;

@Entity
@Table(name = "vaccinations")
@Data
public class Vaccination {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "pet_id", nullable = false)
    private Long petId;
    
    @Column(nullable = false)
    private String name;
    
    @Column(name = "vaccination_date")
    private LocalDate vaccinationDate;
    
    @Column(name = "next_due_date")
    private LocalDate nextDueDate;
    
    @Column(name = "reminder_method")
    private String reminderMethod;
    
    private String veterinarian;
    
    private String hospital;
    
    @Column(columnDefinition = "TEXT")
    private String notes;
}

File: backend-python/app/models/vaccination.py
from sqlalchemy import Column, Integer, String, Date, Text
from app.database import Base

class Vaccination(Base):
    __tablename__ = "vaccinations"
    
    id = Column(Integer, primary_key=True, index=True)
    pet_id = Column(Integer, nullable=False, index=True)
    name = Column(String, nullable=False)
    vaccination_date = Column(Date)
    next_due_date = Column(Date)
    reminder_method = Column(String)
    veterinarian = Column(String)
    hospital = Column(String)
    notes = Column(Text)
